render/egl.c

```diff
@@ -343,6 +337,52 @@ EGLImage wlr_egl_create_image_from_dmabuf(struct wlr_egl *egl,
 		attribs[atti++] = EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT;
 		attribs[atti++] = attributes->modifier[0] >> 32;
 	}
+
+	if (attributes->n_planes > 0) {
```

Comment:
Would prefer to do this in some kind of loop rather than as 3 if-statements with copypasta'd bodies

Reply:
I thought the same but since the constants vary (they have the plane number included) I went that way. Otherwise we'd have to setup all 5 constants upfront with 5 arrays:

pitch_ex_consts[] = [ EGL_DMABUF_PLANE0_PITCH_EX, EGL_DMABUF_PLANE1_EX, ...]

(same for fd, offset, modifier_lo, modifier_hi) and loop over these too or use some macro magic so I went for the open coded version.

Reply:
I'd rather set up the array like you suppose.